prog/vm/host_nexus.rb

```diff
@@ -38,6 +38,27 @@ def before_run
   end
 
   label def start
+    hop_setup_ssh_keys
+  end
+
+  label def setup_ssh_keys
+    ssh_key = Net::SSH::Authentication::ED25519::PrivKey.read(Config.hetzner_ssh_private_key, Config.hetzner_ssh_private_key_passphrase).sign_key
+    init_sshable = Sshable.new(host: sshable.host, unix_user: "root", raw_private_key_1: ssh_key.keypair)
+
+    sshable.update(raw_private_key_1: SshKey.generate.keypair) if sshable.raw_private_key_1.nil?
+    public_key = sshable.keys.first.public_key
+
+    # Replace the key in production, append in other environments.
+    if Config.production?
+      init_sshable.cmd("echo '#{public_key}' > /root/.ssh/authorized_keys")
+    else
+      init_sshable.cmd("echo '#{public_key}' >> /root/.ssh/authorized_keys")
+    end
```

Comment:
We can use `Util.rootish_ssh` instead of new sshable
https://github.com/ubicloud/ubicloud/blob/2437c4c0fa564d227e61b0dd9ecbbfbcde61df0c/prog/bootstrap_rhizome.rb#L21

```suggestion
    if Config.hetzner_ssh_private_key
        sshable.update(raw_private_key_1: SshKey.generate.keypair) unless sshable.raw_private_key_1
        root_key = Net::SSH::Authentication::ED25519::PrivKey.read(Config.hetzner_ssh_private_key, Config.hetzner_ssh_private_key_passphrase).sign_key
        public_key = sshable.keys.first.public_key
        # We don't have `Config.operational_public_keys`, but it might beneficial to have to automate another step. Also it removes "Replace the key in production, append in other environments" step since developer can put his own SSH key to in the dev env
        public_key += "\n#{Config.operational_public_keys}" if Config.operational_public_keys
        Util.rootish_ssh(sshable.host, user, root_key.private_key, "echo '#{public_key.shellescape}' > /root/.ssh/authorized_keys")
    end
```

Reply:
I don't have context as much as @byucesoy, I wrote it as a prototype

Reply:
This makes sense, removing `sshable` simplifies things a lot!
qq on the `operational_public_keys` config. Do we still need to wrap adding those keys under a `production?` check, to prevent adding operational keys to production hosts?

Reply:
We currently add whole team's ssh keys, which is protected with Yubikey, to production hosts manually during setup. Automating this process will help eliminate this step.